### PR TITLE
Improve isElementScrollable snippet

### PR DESCRIPTION
**Pull request by tourdedave**
[Trello](https://trello.com/c/XltNVXmS)

Currently we just check an element's scroll width/height against client width/height. This covers a majority of cases, but there are times when this triggers a false positive. By also checking the element overflow x/y attributes we can have a higher degree of coverage.

This attribute check should be OK because in screenshoter we hide the scrollbars leading up to a target element, but only hide the scrollbars of the target element after doing a lookup of it and checking to see if it is scrollable (e.g.,[here](https://github.com/applitools/eyes.sdk.javascript1/blob/ed01b4d1e0c792baa198334b6f9078be5e733304/packages/screenshoter/src/take-screenshot.js#L51) and [here](https://github.com/applitools/eyes.sdk.javascript1/blob/ed01b4d1e0c792baa198334b6f9078be5e733304/packages/screenshoter/src/getTarget.js#L30))

### CODE REVIEW

**Comment by SophieDePaula** [comment]
@tourdedave looks good to me, except:

1) Need to add a test
2) why did you decide to not add anything to pending changes if this fixes a bug?

**Comment by tourdedave** [comment]
@SophieDePaula Yes, of course. I was going to rely on coverage tests but should add a test for the specific case this fixes (and add an entry to pending changes). I just wanted to see if there were any obvious issues with this approach that y'all might be aware of.

It looks like this change isn't sufficient as-is though, based on what I'm seeing in the coverage tests. I'll keep working on it.

```
  10 failing

  1) Coverage Tests - CheckRegionBySelectorInFrameFullyWithCssStitching
       check region by selector in frame fully with css stitching:
     Error: Test 'TestCheckRegionInFrame' of 'Eyes Selenium SDK - Classic API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133510146?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameFullyWithCssStitching.spec.js:47:5)

  2) Coverage Tests - CheckRegionBySelectorInFrameFullyWithCssStitchingClassic
       check region by selector in frame fully with css stitching classic:
     Error: Test 'TestCheckRegionInFrame' of 'Eyes Selenium SDK - Classic API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133507588?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameFullyWithCssStitchingClassic.spec.js:49:5)

  3) Coverage Tests - CheckRegionBySelectorInFrameFullyWithScrollStitching
       check region by selector in frame fully with scroll stitching:
     Error: Test 'TestCheckRegionInFrame_Scroll' of 'Eyes Selenium SDK - Classic API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133506484?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameFullyWithScrollStitching.spec.js:48:5)

  4) Coverage Tests - CheckRegionBySelectorInFrameFullyWithScrollStitchingClassic
       check region by selector in frame fully with scroll stitching classic:
     Error: Test 'TestCheckRegionInFrame_Scroll' of 'Eyes Selenium SDK - Classic API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133501400?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameFullyWithScrollStitchingClassic.spec.js:49:5)

  5) Coverage Tests - CheckRegionBySelectorInFrameFullyOnFirefoxLegacy
       check region by selector in frame fully on firefox legacy (@jsonwire @firefox):
     Error: Test 'TestCheckRegionInFrame_Scroll' of 'Eyes Selenium SDK - Classic API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133499697?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameFullyOnFirefoxLegacy.spec.js:49:5)

  6) Coverage Tests - CheckRegionBySelectorInFrameMultipleTimesWithCssStitching
       check region by selector in frame multiple times with css stitching:
     Error: Test 'TestCheckRegionInFrame2_Fluent' of 'Eyes Selenium SDK - Fluent API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133490567?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameMultipleTimesWithCssStitching.spec.js:87:5)

  7) Coverage Tests - CheckRegionBySelectorInFrameMultipleTimesWithScrollStitching
       check region by selector in frame multiple times with scroll stitching:
     Error: Test 'TestCheckRegionInFrame2_Fluent_Scroll' of 'Eyes Selenium SDK - Fluent API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133488248?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/CheckRegionBySelectorInFrameMultipleTimesWithScrollStitching.spec.js:87:5)

  8) Coverage Tests - ShouldHideAndRestoreScrollbarsWithCssStitching
       should hide and restore scrollbars with css stitching:
     Error: Test 'TestScrollbarsHiddenAndReturned_Fluent' of 'Eyes Selenium SDK - Fluent API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133178205?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/ShouldHideAndRestoreScrollbarsWithCssStitching.spec.js:49:5)

  9) Coverage Tests - ShouldHideAndRestoreScrollbarsWithScrollStitching
       should hide and restore scrollbars with scroll stitching:
     Error: Test 'TestScrollbarsHiddenAndReturned_Fluent_Scroll' of 'Eyes Selenium SDK - Fluent API' detected differences! See details at: https://eyes.applitools.com/app/batches/00000251753133762896/00000251753133174736?accountId=xIpd7EWjhU6cjJzDGrMcUw~~
      at Eyes.close (/Users/me/Documents/_dev/applitools/eyes.sdk.javascript1/packages/eyes-api/dist/Eyes.js:250:23)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (test/generic/ShouldHideAndRestoreScrollbarsWithScrollStitching.spec.js:49:5)

  10) Coverage Tests - ShouldSendDomAndLocationWhenCheckRegionBySelectorFully
       should send dom and location when check region by selector fully:
     AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

+ undefined
- 'true'
      at Context.<anonymous> (test/generic/ShouldSendDomAndLocationWhenCheckRegionBySelectorFully.spec.js:77:12)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by tourdedave** [comment]
Fixed the test failures. I'll work on adding a test for the case that this fixes.

**Comment by tourdedave** [comment]
@SophieDePaula ready for review

**Review by SophieDePaula — APPROVED** [review]
looks good to me :+1:

File: packages/snippets/src/browser/isElementScrollable.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
+const getElementComputedStyleProperties = require('./getElementComputedStyleProperties')
+
 function isElementScrollable([element] = []) {
-  return element.scrollWidth > element.clientWidth || element.scrollHeight > element.clientHeight
+  const p = getElementComputedStyleProperties([element, ['overflow-x', 'overflow-y', 'overflow']])
+  return (
+    ((p[0] === 'scroll' || p[0] === 'auto') && element.scrollWidth > element.clientWidth) ||
```
Comment (amitzur) [inline packages/snippets/src/browser/isElementScrollable.js:6]: I don't think you need the external parens.
All the `||` conditions are flat (`scrollable horizontally || scrollable vertically || ...`).

File: packages/snippets/src/browser/isElementScrollable.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+    ((p[1] === 'scroll' || p[1] === 'auto') && element.scrollHeight > element.clientHeight) ||
```
Comment (amitzur) [inline packages/snippets/src/browser/isElementScrollable.js:7]: What about `hidden` for `overflow-x` and `overflow-y`?

Comment (amitzur) [inline packages/snippets/src/browser/isElementScrollable.js:7]: What about `hidden` for `overflow-x` and `overflow-y`?

File: packages/snippets/src/browser/isElementScrollable.js [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+    (p[2] === 'hidden' && element.scrollHeight > element.clientHeight)
```
Comment (amitzur) [inline packages/snippets/src/browser/isElementScrollable.js:8]: if you used `hidden` for the specific axes, then this wouldn't be needed, right?

Comment (amitzur) [inline packages/snippets/src/browser/isElementScrollable.js:8]: if you used `hidden` for the specific axes, then this wouldn't be needed, right?

**Review by amitzur — COMMENTED** [review]
I'm so happy to see that the snippets are tested on multiple browsers. But what about firefox?

**Comment by amitzur** [comment]
@benbabayoff8 please review changelog changes

**Comment by benbabayoff8** [comment]
@SophieDePaula @amitzur I would say only that we fixed, without the incorrect fallback part.

File: packages/snippets/src/browser/isElementScrollable.js (hunk 2 above)
Comment (tourdedave) [reply packages/snippets/src/browser/isElementScrollable.js:7]: dup comment, marking as resolved

**Comment by tourdedave** [comment]
I added Firefox coverage for this particular snippet. I attempted to quickly add it to the others, but saw a handful of failures. Seems like a little bit of effort to do it.

**Pull request merged**